build-tools/packages/build-infrastructure/README.md

```diff
@@ -0,0 +1,235 @@
+
+### Loading old config formats
+
+The `repoPackages` configuration currently used by fluid-build will be loaded if the newer `repoLayout` config can't be
+found. This is for back-compat only and will not be maintained indefinitely. Users should convert to `repoLayout` when
+possible.
+
+## Configuration
+
+Configuration for the repo layout is stored in a config file at the root of the repo. This can either be part of the
+`fluidBuild.config.cjs` file in the `repoLayout` property, or in an independent config file named
+`repoLayout.config.cjs` (or mjs).
+
+### Example
+
+The following example configures three workspaces demonstrating the three archetypes - a workspace with multiple release
+groups, a workspace with a single release group that contains multiple packages, and a workspace with a single release
+group that contains a single package.
+
+```js
+repoLayout: {
+  workspaces: {
+    // This is the name of the workspace which is how it's referenced in the API.
+    "client": {
+      // This workspace is rooted at the root of the Git repo.
+      directory: ".",
+      releaseGroups: {
+        // This key is the name of the release group. It must be unique across all
+        // release groups.
+        client: {
+          // The include property can contain package names OR package scopes. If
+          // a scope is provided, all packages with that scope will be a part of
+          // the release group.
+          include: [
+            // Include all the Fluid Framework scopes and packages except for
+            // @fluid-example.
+            "@fluidframework",
+            "@fluid-experimental",
+            "@fluid-internal",
+            "@fluid-private",
+            "@fluid-tools",
+            "@types/jest-environment-puppeteer"
+            "fluid-framework",
+          ],
+          // A release group can have an OPTIONAL root package. This package
+          // is typically private and is similar to the root package for a workspace.
+          // This release group root package may be useful to store scripts or other
+          // configuration only applies on the release group,
+          rootPackageName: "client-release-group-root",
+          defaultInterdependencyRange: "workspace:~",
+
+          // A release group may have an ADO pipeline URL associated with it. This
+          // URL is used to provide direct links to the pipeline when running releases.
+          adoPipelineUrl:
+            "https://dev.azure.com/fluidframework/internal/_build?definitionId=12",
+        },
+        examples: {
+          // This release group contains only the @fluid-example packages.
+          include: ["@fluid-example"],
+          rootPackageName: "examples-release-group-root",
+          defaultInterdependencyRange: "workspace:~",
+        },
+        // If any packages in the workspace don't match a release group, loading the
+        // repo layout config will throw an error.
```

Comment:
So we have to put @fluid-example in a release group in this config even if we don't intend to ever publish them, correct? (ignoring table-document for a moment).

Also, would it make sense to remove the root package name for examples, since they're just a folder within the client workspace?

Reply:
Yes, all packages _must_ be in a release group.

> Also, would it make sense to remove the root package name for examples, since they're just a folder within the client workspace?

I think that file was from another change and shouldn't be in this one. I originally added it to demonstrate what a
release group that wasn't also a workspace might look like, but there's now a test repo so that isn't needed.

Release group root packages are optional, but can be useful to store scripts that are tuned to apply to only that
release group. But that's something we can decide later as we carve up the repo.